src/search-modal/SearchResult.jsx

```diff
@@ -1,37 +1,182 @@
-      <Highlight attribute="content.html_content" hit={hit} />
-      <Highlight attribute="content.capa_content" hit={hit} />
-    </div>
-    <div className="text-muted x-small">
-      {hit.breadcrumbs.map((bc, i) => (
-        // eslint-disable-next-line react/no-array-index-key
-        <span key={i}>{bc.display_name} {i !== hit.breadcrumbs.length - 1 ? '/' : ''} </span>
-      ))}
-    </div>
-  </div>
+const CustomHighlight = ({ attribute, hit, separator }) => (
+  <Highlight
+    attribute={attribute}
+    hit={hit}
+    separator={separator}
+    highlightedTagName="b"
+  />
 );
 
+const ItemIcon = {
+  vertical: Folder,
+  sequential: Folder,
+  chapter: Folder,
+  problem: Question,
+  video: Videocam,
+  html: TextFields,
+};
+
+/**
+ * A single search result (row), usually represents an XBlock/Component
+ * @type {React.FC<{ hit: CustomHit, closeSearch?: () => void}>}
+ */
+const SearchResult = ({ hit, closeSearch }) => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const { libraryAuthoringMfeUrl, redirectToLibraryAuthoringMfe } = useSelector(getStudioHomeData);
+
+  /**
+   * Returns the URL for the context of the hit
+   * @param {CustomHit} hit
+   * @param {boolean?} newWindow
+   * @param {string} libraryAuthoringMfeUrl
+   * @returns {string?}
+   */
+  const getContextUrl = useCallback((newWindow) => {
+    const { context_key: contextKey, usage_key: usageKey } = hit;
+    if (contextKey.startsWith('course-v1:')) {
+      const courseSufix = `course/${contextKey}?show=${encodeURIComponent(usageKey)}`;
+      if (newWindow) {
+        return `${getPath(getConfig().PUBLIC_PATH)}${courseSufix}`;
+      }
+      return `/${courseSufix}`;
+    }
+    if (usageKey.startsWith('lb:')) {
+      if (redirectToLibraryAuthoringMfe) {
+        return `${libraryAuthoringMfeUrl}library/${contextKey}`;
+      }
+    }
+
+    // No context URL for this hit
+    return undefined;
+  }, [libraryAuthoringMfeUrl, redirectToLibraryAuthoringMfe]);
+
+  const redirectUrl = useMemo(() => getContextUrl(), [libraryAuthoringMfeUrl, redirectToLibraryAuthoringMfe]);
+  const newWindowUrl = useMemo(() => getContextUrl(true), [libraryAuthoringMfeUrl, redirectToLibraryAuthoringMfe]);
+
+  /**
+    * Opens the context of the hit in a new window
+    * @param {React.MouseEvent} e
+    * @returns {void}
+    * */
+  const openContextInNewWindow = (e) => {
+    e.stopPropagation();
+
+    if (!newWindowUrl) {
+      return;
+    }
+
+    window.open(newWindowUrl, '_blank');
+  };
+
+  /**
+    * Navigates to the context of the hit
+    * @param {(React.MouseEvent | React.KeyboardEvent)} e
+    * @returns {void}
+    * */
+  const navigateToContext = (e) => {
+    e.stopPropagation();
+
+    if (!redirectUrl) {
+      return;
+    }
+
+    // @ts-ignore Cannot use 'instanceof' with React.KeyboardEvent to narrow down the type
+    if (e.nativeEvent instanceof KeyboardEvent && e.key !== 'Enter' && e.key !== ' ') {
```

Comment:
```suggestion
    if ('key' in e && e.key !== 'Enter' && e.key !== ' ') {
```
That's because you're narrowing the type of `e.nativeEvent`, not `e`. Just use this ^^ and it works without `ts-ignore` :)

Reply:
Yep! Checking `e` against `React.KeyboardEvent` at runtime was throwing an error.
Thank you for the fix!